pkgs/build-support/rust/hooks/cargo-build-hook.sh

```diff
@@ -32,4 +32,6 @@ cargoBuildHook() {
     echo "Finished cargoBuildHook"
 }
 
-buildPhase=cargoBuildHook
+if [ -z "${dontUseCargoBuild-}" ] && [ -z "${buildPhase-}" ]; then
```

Comment:
Name bikeshedding: `dontCargoBuild`? To bring it in line with e.g. `dontConfigure`, `dontBuild`, etc.

Reply:
I was modelling it after the python hooks which use `dontUse`.

cc @FRidh

Reply:
The Python naming is structured but not very pleasant. Given we lack structured attributes, I think the structure is, at least up to a degree, needed. Maybe with the Python hooks we can drop `use`, I don't know.

Reply:
I think the meson hook also does a `dontUseMesonConfigure`. Not very pleasant, but I guess it's used in more than one place and isn't too bad.